frontend/components/product-list/ProductListView.tsx

```diff
@@ -47,78 +60,115 @@ export function ProductListView({
-                  )}
-                  <FilterableProductsSection />
-                  {productList.sections.map((section, index) => {
-                     switch (section.type) {
-                        case ProductListSectionType.Banner: {
-                           return (
-                              <BannerSection
-                                 key={index}
-                                 title={section.title}
-                                 description={section.description}
-                                 callToActionLabel={section.callToActionLabel}
-                                 url={section.url}
-                              />
-                           );
-                        }
-                        case ProductListSectionType.RelatedPosts: {
-                           const tags = [productList.title].concat(
-                              section.tags
-                                 ?.split(',')
-                                 .map((tag) => tag.trim()) || []
-                           );
-                           return (
-                              <RelatedPostsSection key={index} tags={tags} />
-                           );
-                        }
-                        case ProductListSectionType.FeaturedProductList: {
-                           const { productList } = section;
-                           if (productList) {
+                  <InstantSearch
+                     searchClient={algoliaClientRef.current}
+                     indexName={indexName}
+                     routing={{
+                        stateMapping: {
+                           stateToRoute(uiState) {
+                              const indexUiState = uiState[indexName];
+                              const routeObject: any = {
+                                 q: indexUiState.query,
+                                 p: indexUiState.page,
+                              };
+                              routeObject.filter = indexUiState.refinementList;
+                              routeObject.range = indexUiState.range;
+                              return routeObject;
+                           },
+                           routeToState(routeState: any) {
+                              return {
+                                 [indexName]: {
+                                    query: routeState.q,
+                                    page: routeState.p,
+                                    refinementList: routeState.filter,
+                                    range: routeState.range,
+                                 },
+                              };
+                           },
+                        },
+                        router: history({
+                           getLocation() {
+                              if (typeof window === 'undefined') {
+                                 return new URL(url!) as unknown as Location;
```

Comment:
This doesn't seem to be valid JS syntax that I'm aware of:

![image](https://user-images.githubusercontent.com/26855/172915849-b1a86063-23fc-47af-be7f-7d26f4eab1eb.png)

Reply:
Oh, that's typescript aparently. Hmm... doesn't seem necessary given that the type of `url` is already `string` and not `string|null`.

Reply:
You're right 👍